src/Bicep.Core/TypeSystem/TypeValidator.cs

```diff
@@ -301,7 +303,20 @@ private static TypeSymbol NarrowObjectType(ITypeManager typeManager, ObjectSynta
 
             if (missingRequiredProperties.Any())
             {
-                diagnostics.Add(DiagnosticBuilder.ForPosition(expression).MissingRequiredProperties(ShouldWarn(targetType), missingRequiredProperties));
+                TextSpan span = expression.Span;
+                var parent = typeManager.GetParent(expression);
+
+                if (parent is ObjectPropertySyntax p1)
+                {
+                    span = p1.Key.Span;
+                }
+                else if (parent is IDeclarationSyntax p2)
+                {
+                    span = p2.Keyword.Span;

```

Comment:
I think it might be clearer to show the diagnostic over the identifier name rather than the keyword. You should be able to access this with `p2.Name`.

That way, instead of:
```
module myMod '...' = {
~~~~~~
```

The highlight should appear as:
```
module myMod '...' = {
       ~~~~~
```

Reply:
+1

Reply:
+1

Reply:
done